src/Controller/BookController.php

```diff
@@ -84,16 +84,48 @@ public function getBooksByTitle(string $title): Response
 
   public function getBooksByCategory(string $category): Response
   {
+    $books = $this->bookDBService->searchBookByCategory($category);
+    if ($books) {
+      return $this->json(['totalItems' => count($books), 'data' => $books]);
```

Comment:
does it work without that serializer- thing?

Reply:
yes, since i changed our dataschema we no longer have a direct circular relationship